check_pgactivity

```diff
@@ -542,6 +542,50 @@ sub find_hostkey($) {
     return "binary defaults";
 }
 
+# Record on disk some given data to a given file.
+#
+# To avoid data loss, the sub:
+# * first require an exclusive lock on the lock file,
+# * load the existing data
+# * call the closure to record its data in there
+# * write back the edited data to disk
+# * close the file and release the lock.
+#
+# The closure is reponsible to define how and where in the structure it wants to
+# record its data.
+#
+# Parameters are :
+#  * $fn_record: a closure responsible to record its data in the given
+#                structure ref.
+#  * $storage: file to write to.
+sub save_internal($$) {
+    my $fn_record = shift;
+    my $storage   = shift;
+    my $all       = {};
+    my $lockfile  = "${storage}.lock";
+
+    open my $fh, '>', $lockfile or die "can't open «${lockfile}»: $!";
+    flock($fh, LOCK_EX) or die "can't get exclusive lock on «${lockfile}»: $!";
+
+    if ( -e $storage ) {
+
+        die "can not write to binary file «${storage}»" unless -w ${storage};
+        die "file «${storage}» not recognized as a check_pgactivity binary file"
+            unless is_storable $storage;
+
+        eval { $all = retrieve($storage) };
+        die "could not retrieve data from «${storage}»:\n  $@" if $@;
+    }
+
+    $fn_record->($all);
+
+    eval { store( $all, $storage ); };
+    die "could not update data in «${storage}»:\n  $@" if $@;
+
+    # closing the fh removes the lock
```

Comment:
nitpicking (and not this patch fault but still): s/removes/releases/

Reply:
done and squashed